src/main/java/ua/com/conductor/security/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package ua.com.conductor.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import ua.com.conductor.service.UserService;
+
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+    private final UserService userService;
+
+    @Autowired
+    public UserDetailsServiceImpl(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String s) throws UsernameNotFoundException {
+        ua.com.conductor.model.User user = userService.findByEmail(s).get();
```

Comment:
```suggestion
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        ua.com.conductor.model.User user = userService.findByEmail(s).get(); // throw here UsernameNotFoundException instead of NoSuchElement
```